script/company.js

```diff
@@ -0,0 +1,138 @@
+function searchParam() {
+    let companyUrl = null;
+    companyUrl = window.location.search;
+    const urlParams = new URLSearchParams(companyUrl);
+    const symbol = urlParams.get('symbol');
+
+    return symbol;       
+};
+class Company {
+    constructor (symbol) {
+        this.companySymbol = symbol;
+        this.searchUrl = `https://stock-exchange-dot-full-stack-course-services.ew.r.appspot.com/api/v3/company/profile/${this.companySymbol}`;
+        this.CompanyProfile = this.getCompanyProfile();   
+    };  
+       
+    async getCompanyProfile(){
+
+        try {
+            this.setIsLoading(true);
+            const url = this.searchUrl
+
+            const response = await fetch(url);
+            const getResults = await response.json();
+            const companyObject = new CompanyObject(
+                getResults.profile, 
+                this.companySymbol,
+                );
+    
+            return companyObject;     
+        
+         } catch(error) {
+
+            return error;
+
+        } finally {
+            this.setIsLoading(false); 
+        };
+    };
+
+    setIsLoading(isLoading) {
+        const spinner = document.getElementById('spinner').style;
+        if (isLoading) {
+            return  spinner.display = "block" ;
+        } else spinner.display = "none";
+    };
+};
+class CompanyObject {
+    constructor (companyObject, companySymbol) {
+        this.companyName = companyObject.companyName;
+        this.description = companyObject.description;
+        this.image = companyObject.image;
+        this.symbol = companySymbol;
+        this.price = companyObject.price;
+        this.changesPercentage = companyObject.changesPercentage;
+        this.website = companyObject.website; 
+        this.printIt = this.print();        
+    };
+    
+    print() {
```

Comment:
Is the function printing something or creating an element in your website. 
Think of a better naming